openquake/risklib/asset.py

```diff
@@ -860,14 +860,16 @@ def _read_csv(self):
             with open(fname, encoding='utf-8') as f:
                 fields = next(csv.reader(f))
                 header = set(fields)
+                missing = expected_header - header - {'exposure', 'country'}
                 if len(header) < len(fields):
                     raise InvalidFile(
                         '%s: The header %s contains a duplicated field' %
                         (fname, header))
-                elif expected_header - header - {'exposure', 'country'}:
-                    raise InvalidFile(
-                        'Unexpected header in %s\nExpected: %s\nGot: %s' %
-                        (fname, sorted(expected_header), sorted(header)))
+                elif missing:
+                    msg = ('Unexpected header in %s\nExpected: %s\nGot: %s\n'
```

Comment:
Here, it would be helpful to list _only_ the field names that do not match, perhaps using set subtract.

Reply:
`missing = expected_header - header - {'exposure', 'country'}` is using set subtract and it is displaying such names